tmt/steps/execute/__init__.py

```diff
@@ -596,16 +721,25 @@ def prepare_tests(self, guest: Guest, logger: tmt.log.Logger) -> list[TestInvoca
 
     def prepare_scripts(self, guest: "tmt.steps.provision.Guest") -> None:
         """ Prepare additional scripts for testing """
+        # For rpm-ostree based distributions use a different default destination directory
+        scripts_dest_dir = effective_scripts_dest_dir(
+            default=SCRIPTS_DEST_DIR_OSTREE if guest.facts.is_ostree else SCRIPTS_DEST_DIR)
+
+        # Create scripts directory
+        guest.execute(Command("mkdir", "-p", str(scripts_dest_dir)))
+
         # Install all scripts on guest
         for script in self.scripts:
-            source = SCRIPTS_SRC_DIR / script.path.name
-
-            for dest in [script.path, *script.aliases]:
-                guest.push(
-                    source=source,
-                    destination=dest,
-                    options=["-p", "--chmod=755"],
-                    superuser=guest.facts.is_superuser is not True)
+            with script as source:
+                for dest in [script.source_filename, *script.aliases]:
```

Comment:
`dest` is now just a filename, correct? `filename` would be better here, there's way too many "dest"-ish names already.

Reply:
yeah, it is, fixed